game-core/src/main/java/games/strategy/triplea/attachments/TechAbilityAttachment.java

```diff
@@ -459,19 +388,12 @@ public int getMinimumTerritoryValueForProductionBonus() {
   }
 
   public static int getMinimumTerritoryValueForProductionBonus(final PlayerID player, final GameData data) {
-    int minimumTerritoryValueForProductionBonus = -1;
-    for (final TechAdvance ta : TechTracker.getCurrentTechAdvances(player, data)) {
-      final TechAbilityAttachment taa = TechAbilityAttachment.get(ta);
-      if (taa != null) {
-        final int min = taa.getMinimumTerritoryValueForProductionBonus();
-        if (min == -1) {
```

Comment:
Are undefined values being ignored below as this condition suggests they should?

Reply:
Right, I thoght about this, but I thought it wasn't needed, my bad.